src/providers/ses.rs

```diff
@@ -44,44 +46,70 @@ impl Provider for SesProvider {
         &self,
         to: &str,
         cc: &[&str],
+        headers: Option<&Headers>,
         subject: &str,
         body_text: &str,
         body_html: Option<&str>,
     ) -> Result<String, ProviderError> {
-        let mut destination = Destination::default();
-        destination.to_addresses = Some(vec![to.to_string()]);
+        let mut all_headers = header::Headers::new();
+        all_headers.set(header::From(vec![self.sender.parse::<Mailbox>()?]));
+        all_headers.set(header::To(vec![to.parse::<Mailbox>()?]));
+        all_headers.set(header::Subject(subject.into()));
+        all_headers.set(header::MIME_VERSION_1_0);
+
         if cc.len() > 0 {
-            destination.cc_addresses = Some(cc.iter().map(|s| s.to_string()).collect());
+            let mut parsed_cc: Vec<Mailbox> = Vec::new();
+            for address in cc.iter() {
+                parsed_cc.push(address.parse::<Mailbox>()?)
+            }
+            all_headers.set(header::Cc(parsed_cc))
+        }
+
+        if let Some(headers) = headers {
+            for (name, value) in headers {
+                all_headers.append_raw(name.to_owned(), value.to_owned())
+            }
         }
 
-        let mut message = Message::default();
-        message.subject.data = subject.to_string();
-        message.body.text = Some(Content {
-            charset: None,
-            data: body_text.to_string(),
-        });
-        if let Some(html) = body_html {
-            message.body.html = Some(Content {
-                charset: None,
-                data: html.to_string(),
-            });
+        let mut message: Message<MultiPart<String>> = Message::new().with_headers(all_headers);
+
+        let mut body: MultiPart<String> = MultiPart::alternative().with_singlepart(
+            SinglePart::quoted_printable()
+                .with_header(header::ContentType(
+                    "text/plain; charset=utf8".parse().unwrap(),
```

Comment:
Just noting for future reference, when we open this API up to external callers we should make it explicit in the docs that UTF-8 is assumed (or provide a param that lets them override it).